src/components/shared/thread/SelectedActivityGraph.js

```diff
@@ -1,271 +0,0 @@
-import type { Viewport } from '../chart/Viewport';
-
-type CanvasProps = {|
-  // The viewport property is injected by the withViewport component, but is not
-  // actually used or needed in this case. However, withViewport has side effects
-  // of enabling event listeners for adjusting the view.
-  +viewport: Viewport,
-  +selectedThreadIndex: ThreadIndex,
-  +interval: Milliseconds,
-  +rangeStart: Milliseconds,
-  +rangeEnd: Milliseconds,
-  +fullThread: Thread,
-  +filteredThread: Thread,
-  +callNodeInfo: CallNodeInfo,
-  +selectedCallNodeIndex: IndexIntoCallNodeTable | null,
-  +categories: CategoryList,
-  +samplesSelectedStates: null | SelectedState[],
-  +treeOrderSampleComparator: (
-    IndexIntoSamplesTable,
-    IndexIntoSamplesTable
-  ) => number,
-  +selectBestAncestorCallNodeAndExpandCallTree: typeof selectBestAncestorCallNodeAndExpandCallTree,
-  +selectLeafCallNode: typeof selectLeafCallNode,
-  +focusCallTree: typeof focusCallTree,
-|};
-
-class SelectedThreadActivityGraphCanvas extends PureComponent<CanvasProps> {
-  /**
-   *
-   */
-  _onActivitySampleClick = (sampleIndex: IndexIntoSamplesTable) => {
-    const {
-      selectedThreadIndex,
-      selectBestAncestorCallNodeAndExpandCallTree,
-      focusCallTree,
-    } = this.props;
-    selectBestAncestorCallNodeAndExpandCallTree(
-      selectedThreadIndex,
-      sampleIndex
-    );
-    focusCallTree();
-  };
-
-  _onStackSampleClick = (sampleIndex: IndexIntoSamplesTable) => {
-    const {
-      selectedThreadIndex,
-      selectLeafCallNode,
-      focusCallTree,
-    } = this.props;
-    selectLeafCallNode(selectedThreadIndex, sampleIndex);
-    focusCallTree();
-  };
-
-  render() {
-    const {
-      fullThread,
-      filteredThread,
-      interval,
-      rangeStart,
-      rangeEnd,
-      callNodeInfo,
-      selectedCallNodeIndex,
-      categories,
-      samplesSelectedStates,
-      treeOrderSampleComparator,
-    } = this.props;
-
-    return (
-      <div>
-        <ThreadActivityGraph
-          interval={interval}
-          fullThread={fullThread}
-          className="selectedThreadActivityGraph"
-          rangeStart={rangeStart}
-          rangeEnd={rangeEnd}
-          onSampleClick={this._onActivitySampleClick}
-          categories={categories}
-          samplesSelectedStates={samplesSelectedStates}
-          treeOrderSampleComparator={treeOrderSampleComparator}
-        />
-        <ThreadStackGraph
-          interval={interval}
-          thread={filteredThread}
-          className="selectedThreadStackGraph"
-          rangeStart={rangeStart}
-          rangeEnd={rangeEnd}
-          callNodeInfo={callNodeInfo}
-          selectedCallNodeIndex={selectedCallNodeIndex}
-          onSampleClick={this._onStackSampleClick}
-          categories={categories}
-          stacksGrowFromCeiling={true}
-        />
-      </div>
-    );
-  }
-}
-
-const SelectedThreadActivityGraphCanvasWithViewport = withChartViewport(
```

Comment:
I really like the way to zoom and with mouse and shift + scroll. I think we are gonna lose that ability in the profiler since we don't have SelectedThreadActivity anymore, right? Not for this patch probably but should we implement this for the timeline?

Reply:
Yeah, that'd be nice to implement for the timeline!

Reply:
#2398